src/Reports/Full.php

```diff
@@ -64,15 +64,28 @@ public function generateFileReport($report, File $phpcsFile, $showSources=false,
         foreach ($report['messages'] as $line => $lineErrors) {
             foreach ($lineErrors as $column => $colErrors) {
                 foreach ($colErrors as $error) {
-                    $length = strlen($error['message']);
+                    // Start with the presumption of a single line error message.
+                    $length    = strlen($error['message']);
+                    $srcLength = (strlen($error['source']) + 3);
                     if ($showSources === true) {
-                        $length += (strlen($error['source']) + 3);
+                        $length += $srcLength;
+                    }
+
+                    // ... but also handle multi-line messages correctly.
+                    if (strpos($error['message'], "\n") !== false) {
+                        $errorLines = explode("\n", $error['message']);
+                        $length     = max(array_map('strlen', $errorLines));
+
+                        if ($showSources === true) {
+                            $lastLine = array_pop($errorLines);
```

Comment:
It looks like we are only using the length of the last line of the error message inside this.  If `$error['message'] = "... \n ... \nthe end."` and the $showSources flag is true, we would choose the max between:

1. $length ( length of the longest line in multi-line error message ) 
2. strlen("the end.") + $srcLength

the length of the lonegest line(option 1) does not have `+ $srcLength ` included because it uses the new value of $length in line 77

Reply:
Let's take a slightly more concrete example,
```
$error['message'] = "First line of the message.\nSecond line of the message. This line is something long.\nthe end.";
```

What you're misunderstanding is that the output in that case would be something like
```
 0 | ERROR | First line of the message.
   |       | Second line of the message. This line is something long.
   |       | the end. (Standard.Category.Sniff.Message)
```
i.e. the source is always added to the last line.

Reply:
@elstamey Thanks for looking this over, the code, however, is correct.

This code is used to find the longest line by looping over all error messages which need to be added to the report.
The code block we're looking at is where the line length of one individual error message is determined.

The error source will only ever be added to the last line of an individual error message.

A message can either be single line, in which case, the code uses the length of the message + length of the source code as `$length`.

Or a message is multi-line, in which case `$length` is determined as follows:
* Find the longest individual line in the error message (line 75).
* Get the last line and add the error code to it, determine the length and overrule the `$length` set on line 75 if the last line including the error code would be longer than the longest individual line in the error message.

After that, the determined `$length` is compared with the the longest line length of all previously seen messages, `$maxErrorLength`, and overruling that if the longest line in the current message would be longer.

Does that make sense now ?

While writing this out, I did think of an additional efficiency tweak which could be made (`break` if `$maxErrorLength` > `$width` as there is no need to continue looking at the rest of the error messages), but I will leave that for a future PR.

Reply:
After looking at some of the other issues and tickets linked to this one, I realized why the last line length is being grabbed when the srcLength is added.  This should have been a bit more obvious to me since the `(Squiz.Commenting.ClosingDeclarationComment.Missing)` appears at the end of the error message.  🧠 

However, I don't understand why it matters if the length of the longest line is longer than the length of the last line + $srcLength.  If #1 > #2, does the $srcLength not matter?  do we not need #1 + $srcLength in our $maxErrorLength (line 85) when we are looping?

Reply:
OK, I am really sorry that my second comment took me so long to finish before you both responded.  Thank you so much for your examples because they helped a lot.  

For what it's worth, I did run your example test above @jrfnl and was able to see the issue and see it fixed, I just kept thinking the omission of $srcLength was going to cause a problem.  😞 

Thank you both again for taking the time to reply!